### ISSUE TITLE
dsh install prerequisites  installs SW I already have

### ISSUE THREAD

**Issue by stkrzysiak** [blinkreaction/drude#114]
Despite having vagrant 1.8.1 on my Mac, dsh install prerequisites is downloading and installing 1.8.1.  This behavior is not ideal, i may have 2/3 prerequisites installed, so it should only install the missing one.  I'd be open to fixing this, unless this is desired behavior and I'm overlooking the reason for it.

**Comment by lmakarov** [comment]
Yep, completely agree.
We should improve the install process and check already installed prerequisites and their versions, and not assume everyone starts from scratch.

**Comment by achekulaev** [comment]
@lmakarov 
We can not do that actually. We install things using homebrew. And update using same. If there's similar software installed not via homebrew that all good and fine but we don't know whether that software is up to date and can't be sure it's gonna work with drude. 

I'd recommend uninstalling version that was installed not with brew.

At the same time if vagrant/whatever_else was installed via homebrew there is no harm in running `brew install vagrant` once again.

**Comment by hanoii** [comment]
I was on BoF on drupalcon and interested in this. I have installed not only vagrant but also docker, docker compose and probably most of the necessary tools. So I wonder if at least, there should not be  way of simply allowing manual install of tools (in whatever method we are comfortable with), with the required versions and allow us to still use drude.

The install could also check the version number and request for a manual upgrade if it cannot do it on its own.

**Comment by achekulaev** [comment]
@hanoii you can still use Drude for sure! `dsh install prerequisites` is only there to **help** you install required software. If you already have latest-ish `VirtualBox`, `vagrant`, `docker` and `docker-compose` then you pretty much have all the required software. 

Simply don't run `dsh install prerequisites` and only run everything else to set up Drude.

**Comment by stkrzysiak** [comment]
@hanoii I think I was sitting next to you at that BoF!  Anyway, I am trying out Kalabox now on windows and it's doing the same thing, re-installing all this software I already have.  This seems to be a theme for me lately.  
Regarding Drude @achekulaev, would drude accept a PR that provides a workflow similar to what @hanoii  described? 

> The install could also check the version number and request for a manual upgrade if it cannot do it on its own.

**Comment by achekulaev** [comment]
@stkrzysiak yes sure, but PR would also need to check that software is not installed with homebrew (if we're talking Mac) because if it was installed with brew then it can be upgraded and there's no need in manuall installation/upgrades.

On Windows there is another packet manager, though. So windows would require it's own different checks. Of course you could only cover Mac in your PR but you see a can of worms it opens, right?

_update:_ I'm not sure why would you need it though. If you don't want to re-install software it's totally fine. Don't re-install it. Don't run `install prerequisites`.

**Comment by stkrzysiak** [comment]
> update: I'm not sure why would you need it though. If you don't want to re-install software it's totally fine. Don't re-install it. Don't run install prerequisites.

But if you have 3/4 installed it should just install the last remaining prerequisite.  Using this flag would still leave you at 3/4 requisites installed, correct?
